### ISSUE TITLE
[TW-1479] Task arguments not properly parsed if they contain a space

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1505]
_Tomas Babej on 2014-12-26T17:08:23Z says:_ 

 I have a task which description contains a space:

```
$ [tbabej] task 13 export
{"id":13,"description":"1.st task","entry":"20141226T165008Z","modified":"20141226T165008Z","status":"pending","uuid":"256721ea-6e02-4385-b39f-ea7b27592262","urgency":"0"}
```

This is 2.4.0 dev. version
```
$ [tbabej] task _version
e924913
$ [tbabej] task --version
2.4.0.beta3
```

When trying to query for the task by description, the following works:

```
$ [tbabej] task 1.st
[task next description ~ '1.st']

ID Age  Recur Description Urg 
13 9min       1.st task      0

1 task
There are local changes.  Sync required.
```

Also, the following works:

```
$ [tbabej] task description:1.st                                                                                                                                                                                                                    
[task next description = 1.st]

ID Age   Recur Description Urg 
13 10min       1.st task      0

1 task
There are local changes.  Sync required.
```

However, if I pass the space as well (by escaping it):
```
$ [tbabej] task description:1.st\ task
[task next description = 1.st task]
The expression could not be evaluated.
```

```
$ [tbabej] task description:"1.st task"                                                                                                                                                                                                                             
[task next description = 1.st task]
The expression could not be evaluated.
```

If I omit the description keyword, the exact match is no longer used, and we have a match:

```
$ [tbabej] task "1.st task"                                                                                                                                                                                                                                         
[task next description ~ '1.st task']

ID Age   Recur Description Urg 
13 10min       1.st task      0

1 task
There are local changes.  Sync required.
```

I think this is inconsistent and we should be able to query for the tasks even with description: prefix.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-12-26T17:08:23Z
Modified: 2015-07-13T16:06:25Z
```

**Comment by taskwarrior** [comment]
_Tomas Babej on 2014-12-26T17:09:05Z says:_ 

 Seems I can't create a issue without a issue :) this one should be 'Core' component.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-12-27T16:25:41Z says:_ 

 These work:

```
$ task '/1.st task/'
$ task '1.st task'
$ task description == "'1.st task'"
```

**Comment by taskwarrior** [comment]
_Tomas Babej on 2015-01-03T22:23:58Z says:_ 

 For filtering by project with a space, there will be no workaround when 'is' modifier is deprecated:

```
task "project == 'random project'" info
The expression could not be evaluated.
```

**Comment by tbabej** [comment]
Regression on 2.5.2, 2.6.0. Related to (or duplicate of) #1641.

**Comment by tbabej** [comment]
Actually this is resolved on 2.6.0.

**Comment by mrossinek** [comment]
Yes same thing here. My commit https://github.com/GothenburgBitFactory/taskwarrior/commit/d85901e7ff0ce0aec8e6090dd601552b08f61007 actually referred to this issue but I didn't include a keyword to actually close it.
